spec/system/tech_shop_spec.rb

```diff
@@ -37,150 +39,142 @@
     # Subproducts
     @camera = Camera.create!(name: 'canon eos rebel t3', href: 'canon')
 
-    100.times { Product.create!(name: 'crapoola', href: 'crappy', tags: ['crappy']) }
-
-    @products_in_database = Product.all
-
     Product.reindex!(MeiliSearch::Rails::IndexSettings::DEFAULT_BATCH_SIZE, true)
   end
 
-  it 'is not synchronous' do
-    p = Product.new
-    p.valid?
+  context 'product' do
+    it 'defaults to asynchronous' do
+      p = Product.new
 
-    expect(p).not_to be_ms_synchronous
-  end
+      expect(p).not_to be_ms_synchronous
+    end
 
-  it 'is able to reindex manually' do
-    results_before_clearing = Product.raw_search('')
-    expect(results_before_clearing['hits'].size).not_to be(0)
-    Product.clear_index!(true)
-    results = Product.raw_search('')
-    expect(results['hits'].size).to be(0)
-    Product.reindex!(MeiliSearch::Rails::IndexSettings::DEFAULT_BATCH_SIZE, true)
-    results_after_reindexing = Product.raw_search('')
-    expect(results_after_reindexing['hits'].size).not_to be(0)
-    expect(results_before_clearing['hits'].size).to be(results_after_reindexing['hits'].size)
+    it 'supports manual indexing' do
+      products_before_clear = Product.raw_search('')['hits']
+      expect(products_before_clear).not_to be_empty
+
+      Product.clear_index!(true)
+
+      products_after_clear = Product.raw_search('')['hits']
+      expect(products_after_clear).to be_empty
+      Product.reindex!(MeiliSearch::Rails::IndexSettings::DEFAULT_BATCH_SIZE, true)
```

Comment:
This syntax is still clunky but I'm saving it for when I refactor the actual lib, not the test suite.